src/main/java/org/springframework/integration/kafka/outbound/KafkaProducerMessageHandler.java

```diff
@@ -108,7 +112,9 @@ protected void handleMessageInternal(final Message<?> message) throws Exception
 				? this.messageKeyExpression.getValue(this.evaluationContext, message)
 				: message.getHeaders().get(KafkaHeaders.MESSAGE_KEY);
 
-		this.kafkaProducerContext.send(topic, partitionId, messageKey, message.getPayload());
+		Future<RecordMetadata> future = this.kafkaProducerContext.send(topic, partitionId, messageKey,
```

Comment:
We do this:
```
if (!producerMetadata.isSync()) {
	return future;
}
```
what can be handled by the provided `ProducerListener`. And with this Gary's concerns I guess we have to use `LoggingProducerListener` as the default one to worry end-user that he should provide his own impl to deal with those errors.
But anyway we must not call `Future.get()` for the `async` mode.